File: Recursion/Find_all_factorial_numbers_less_than_or_equal_to_N.cpp
class Solution
{
public:
    vector<long long> ans;
    long long fact(long long n)
    {
        if (n == 0)
            return 1;
        return n * fact(n - 1);
    }
    vector<long long> factorialNumbers(long long N)
    {
        for (int i = 1; i <= N; i++)
        {
            if (fact(i) <= N)
                ans.push_back(fact(i));
            else
                break;
        }
        return ans;
    }
};

File: Recursion/Find_all_factorial_numbers_less_than_or_equal_to_N.js
let ans = [];
function fact(n) {
  if (n == 0) return 1;
  return n * fact(n - 1);
}

function factorial(n) {
  for (let i = 1; i <= n; i++) {
    if (fact(i) <= n) ans.push(fact(i));
    else break;
  }
  return ans;
}
console.log(factorial(6));
